src/js/components/Box/Box.js

```diff
@@ -54,6 +56,10 @@ const Box = forwardRef(
 
     const skeleton = useSkeleton();
 
+    const responsiveContainer = useContext(ResponsiveContainerContext);
+    const responsive =
+      responsiveContainer && responsiveProp ? 'container' : responsiveProp;
```

Comment:
I'm a little confused about this line, I would expect `responsive={true}` and `responsive='container'` to mean different things. It seems like this line would result in responsive being set to 'container' when the value is `true`.

Reply:
It would. This value is what is passed to the various apis used by the styled components involved (breakpoint css, edge, and round etc.) If `responsiveProp` is true or 'container' (basically not false) this Box needs to do responsive style stuff. If this Box is supposed to be responsive it needs to know whether it's inside a ResponsiveContainerContext (ie some Box way above it in the parentage has `responsive="container"`. The breakpoint style apis need to know this but can't check a context. So this is how we tell them which type of responsive.